metadata-ingestion/src/datahub/ingestion/source/hex/hex.py

```diff
@@ -0,0 +1,161 @@
+        description="Set ownership identity from owner/creator email",
+    )
+
+
+class HexReport(StaleEntityRemovalSourceReport, HexAPIReport):
+    pass
+
+
+@platform_name("Hex")
+@config_class(HexSourceConfig)
+@support_status(SupportStatus.TESTING)
+@capability(SourceCapability.DESCRIPTIONS, "Supported by default")
+@capability(SourceCapability.OWNERSHIP, "Supported by default")
+@capability(SourceCapability.PLATFORM_INSTANCE, "Enabled by default")
+@capability(SourceCapability.CONTAINERS, "Enabled by default")
+@capability(SourceCapability.DELETION_DETECTION, "Enabled via stateful ingestion")
+class HexSource(StatefulIngestionSourceBase):
+    def __init__(self, config: HexSourceConfig, ctx: PipelineContext):
+        super().__init__(config, ctx)
+        self.source_config = config
+        self.report = HexReport()
+        self.platform = HEX_PLATFORM_NAME
+        self.hex_api = HexAPI(
+            report=self.report,
+            token=self.source_config.token.get_secret_value(),
+            base_url=self.source_config.base_url,
+            page_size=self.source_config.page_size,
+        )
+        self.mapper = Mapper(
+            workspace_name=self.source_config.workspace_name,
+            platform_instance=self.source_config.platform_instance,
+            base_url=self.source_config.base_url,
+            patch_metadata=self.source_config.patch_metadata,
+            collections_as_tags=self.source_config.collections_as_tags,
+            status_as_tag=self.source_config.status_as_tag,
+            categories_as_tags=self.source_config.categories_as_tags,
+            set_ownership_from_email=self.source_config.set_ownership_from_email,
+        )
+
+    @classmethod
+    def create(cls, config_dict: Dict[str, Any], ctx: PipelineContext) -> "HexSource":
+        config = HexSourceConfig.parse_obj(config_dict)
+        return cls(config, ctx)
+
+    def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
+        return [
+            *super().get_workunit_processors(),
+            StaleEntityRemovalHandler.create(
+                self, self.source_config, self.ctx
+            ).workunit_processor,
+        ]
+
+    def get_report(self) -> StatefulIngestionReport:
+        return self.report
+
+    def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
+        yield from self.mapper.map_workspace()
+
+        for project_or_component in self.hex_api.fetch_projects():
+            if isinstance(project_or_component, Project):
+                if self.source_config.project_titles_pattern.allowed(
+                    project_or_component.title
+                ):
+                    yield from self.mapper.map_project(project=project_or_component)
+            elif (
+                isinstance(project_or_component, Component)
+                and self.source_config.include_components
+            ):
+                if self.source_config.component_titles_pattern.allowed(
```

Comment:
This is super nitpicky, but I think the singular version reads better

```suggestion
                if self.source_config.component_title_pattern.allowed(
```

Reply:
addressed in commit 65e360672267823769bc2e95a2006e34e4f70399